compiler/optimizer/VPConstraint.cpp

```diff
@@ -4802,8 +4802,8 @@ bool TR::VPClass::mustBeEqual(TR::VPConstraint *other, TR::ValuePropagation *vp)
       return getKnownObject()->getIndex() == other->getKnownObject()->getIndex();
 
    TR::VPClass *otherClass = NULL;
-   if (other)
-      otherClass = other->asClass();
+   otherClass = other->asClass();
```

Comment:
this should be merged with the otherClass declaration - why declare and then set rather than set when declaring?